### ISSUE TITLE
bug: Parsing risc0 proof which is 0 padded does not parse correctly

### ISSUE THREAD

**Issue by stefanMadzharov** [keep-starknet-strange/garaga#230]
# Bug Report

**Garaga version:**

**Current behavior:**

In the moment if we have for example in the `proof.json` the following journal value:
`{
    "journal": "0x00000000000000000000000000000000000000000000000000000000000f3893"
}`
the 
`return Groth16Proof._from_risc0(
   seal=bytes.fromhex(seal[2:]),
   image_id=bytes.fromhex(image_id[2:]),
   journal=bytes.fromhex(journal[2:]),
)` is failing because the line 
`journal=bytes.fromhex(journal[2:]),` is failing and the fallback option is to call
` public_inputs = find_item_from_key_patterns(data, ["public"])` which fails.

**Expected behavior:**

The correct behavior would be to correctly parse the journal.

**Steps to reproduce:**

Just put 
`{
    "journal": "0x00000000000000000000000000000000000000000000000000000000000f3893"
}` in the `proof.json` and it would fail.

**Comment by stefanMadzharov** [comment]
I think I can fix it.

**Comment by feltroidprime** [comment]
Right.
Important finding indeed. 

I would suggest 
- adding a specific Exception to https://github.com/keep-starknet-strange/garaga/blob/584cad66d067aa6c04447fe06a00c7be99bc2445/hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py#L34-L52 to catch specifically this exception and pass only for this one https://github.com/keep-starknet-strange/garaga/blob/584cad66d067aa6c04447fe06a00c7be99bc2445/hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py#L334-L341 
- Changing the logic 

```
            seal = io.to_hex_str(find_item_from_key_patterns(data, ["seal"]))
            image_id = io.to_hex_str(find_item_from_key_patterns(data, ["image_id"]))
            journal = io.to_hex_str(find_item_from_key_patterns(data, ["journal"]))
```

To hex string is trmming the 0's on the left hence it's taking bytes of even number of nibbles is failing.

**Comment by feltroidprime** [comment]
closed by #233
